app/services/provider_authorisation.rb

```diff
@@ -129,23 +133,34 @@ def full_authorisation?(permission:, course:)
     # enforce user-level permissions
     errors << :requires_provider_user_permission unless
       @actor.is_a?(VendorApiUser) ||
-        @actor.provider_permissions.send(permission)
-          .exists?(provider: [training_provider, ratifying_provider].compact)
+        user_level_can?(permission: permission, provider: training_provider) ||
+        user_level_can?(permission: permission, provider: ratifying_provider)
 
     # enforce org-level permissions
     if ratifying_provider.present?
-      if @actor.providers.include?(training_provider)
-        errors << :requires_training_provider_permission unless
-          permission_as_training_provider_user?(
-            permission: permission,
-            course: course,
-          )
+      ratifying_provider_can = permission_as_ratifying_provider_user?(permission: permission, course: course)
+      training_provider_can = permission_as_training_provider_user?(permission: permission, course: course)
+
+      # If user belongs to both providers, usually one of the two has org-level perm.
+      if @actor.providers.include?(ratifying_provider) && @actor.providers.include?(training_provider)
+        # If not, there is something wrong with the setup
+        if [training_provider_can, ratifying_provider_can].none?
+          errors << :requires_training_provider_permission
+          errors << :requires_ratifying_provider_permission
+        # Check org-level and user-level permissions match for ratifying provider
+        elsif !training_provider_can
+          errors << :requires_provider_user_permission unless
+            user_level_can?(permission: permission, provider: ratifying_provider)
```

Comment:
Do we need to check user permissions here again given we did it at the top of the method?

Reply:
I thought the same thing, but I think this is necessary because the above check tests for permissions on either side, and we need to know that the user permissions align — eg above check could be true because you have permissions at the ratifier, but fail here because only the trainer has this permission